### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by Elijahdre**
In this PR, I have:
- Create a class `Nameable`.
   - Implement a method called `correct_name` that will raise a `NotImplementedError`.

#### Turn your Person class to Nameable
- Make sure that your Person class inherits from Nameable
- Make sure that this class has a method `correct_name` implemented. It should simply return the name attribute.

#### Prepare base Decorator
- Make sure that it inherits from Nameable.
- In the constructor assign a nameable object from params to an instance variable.
- Implement the `correct_name` method that returns the result of the correct_name method of the `@nameable`.

#### Prepare CapitalizeDecorator and TrimmerDecorator
- For the CapitalizeDecorator:
    - Create a class that inherits from the base Decorator class.
    - Implement a method `correct_name` that capitalizes the output of `@nameable.correct_name`.
- For the TrimmerDecorator:
    - Create a class that inherits from the base Decorator class.
    - Implement a method `correct_name` that makes sure that the output of `@nameable.correct_name` has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

File: decorators/trimmer_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+require_relative 'decorator'
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name
+      .name[0...10]
+  end
```
Comment (DeliceLydia) [inline decorators/trimmer_decorator.rb]: - [ ] I am getting an error when I run the app that comes from this class:
Screenshot from 2022-11-03 17-01-23
 - you can do 
 ```ruby
def correct_name
      @nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name
    end
```

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @Elijahdre, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




### Cheers!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)

File: decorators/trimmer_decorator.rb (hunk 1 above)
Comment (Elijahdre) [reply decorators/trimmer_decorator.rb]: @DeliceLydia  Thanks!

Comment (Whoistolu) [reply decorators/trimmer_decorator.rb]: Well resolved 👍🏽

**Review by Whoistolu — APPROVED** [review]
### Hi @Elijahdre 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: decorators/trimmer_decorator.rb (hunk 1 above)
Comment (Elijahdre) [reply decorators/trimmer_decorator.rb]: Thanks man @Whoistolu

Comment (Elijahdre) [reply decorators/trimmer_decorator.rb]: Thanks man @Whoistolu

**Comment by Elijahdre** [comment]
@Whoistolu Thank you mate 🙏

**Pull request merged**